### PR TITLE
Use `async`/`await` where applicable

### PR DESCRIPTION
**Pull request by Androbin**
### Background

Needed for #386 @itaim
Needed for #1139 @merwanehamadi 
Needed for #1937 @arrenv
Needed for #1243
Needed for #2442 @SwiftyTheCoder 

### Changes

Use `async`/`await` where applicable, change `openai.{ChatCompletion,Embedding}.create` to `.acreate`

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by Wladastic** [comment]
Sounds great, I think it will also fix my issue where agents start talking asynchronously for some reason.
I hear them talking word for word in a very weird and gibberish way

**Comment by nponeccop** [comment]
@Androbin There are conflicts now

**Comment by Androbin** [comment]
@richbeales PTAL

**Comment by Androbin** [comment]
@Pwuts PTAL

File: main.py [hunk 1]
```diff
@@ -1 +0,0 @@
-from autogpt import main
```
Comment (ntindle) [inline main.py]: ?

**Comment by p-i-** [comment]
Can we put a pin in this one? There's another async PR that I'd prefer to have merged.

https://github.com/Significant-Gravitas/Auto-GPT/pull/2442

Sorry, super-awkward when this happens. I've put a wontfix label on this one for now.

Team is working on a mechanism whereby PRs need to be created as an issue that then is referenced by the PR, so we can avoid dups.

If there's a reason for promoting THIS PR, please do say!

**Comment by Androbin** [comment]
@p-i- I don't see how this PR conflicts with #2442. It does not change any structure, all it does is replace `.create` with `.acreate` and propagate `async` and `await` through the caller hierarchy. Any PR seeking to introduce async code has to do exactly that. If #2442 ends up reimplementing some of these functions, it still needs to add `async` and `await` everywhere else.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Pull request closed without merging**